src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Public.Data/DataSetVersionMappingControllerTests.cs

```diff
@@ -1139,38 +1197,43 @@ await TestApp.AddTestData<PublicDataDbContext>(context =>
         [Fact]
         public async Task NotBauUser_Returns403()
         {
-            var client = BuildApp(user: DataFixture.AuthenticatedUser()).CreateClient();
-
-            var response = await GetFilterMappings(nextDataSetVersionId: Guid.NewGuid(), client: client);
+            var response = await GetFilterMappings(
+                nextDataSetVersionId: Guid.NewGuid(),
+                client: fixture.CreateClient().WithUser(OptimisedTestUsers.Authenticated)
+            );
```

Comment:
Just looking at a potential inconsistency with the tests in this class compared to DataSetCandidatesControllerTests where they only pass the ClaimsPrincipal user to a helper method which creates the client.

Wouldn't it make sense to do the same in this class, i.e. alter `GetFilterMappings`, `GetLocationMappings`, `ApplyBatchLocationMappingUpdates` and `ApplyBatchLocationMappingUpdates` to accept a user parameter and then call `fixture.CreateClient().WithUser` within them, avoiding some of the repitition?

Reply:
I've done this across the board, but also in doing so, I realised that the user pool mechanism was overly complex, and I've now replaced it with a "user holder" instead which just holds the current user that you want to use for the next HTTP request.  Because of this, we can get rid of the mechanism whereby we use an HTTP header to specify a registered user's id, and the whole thing is a lot simpler now.